src/alt-util.cpp

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2019-2020 Xenios SEZC
+// https://www.veriblock.org
+// Distributed under the MIT software license, see the accompanying
+// file LICENSE or http://www.opensource.org/licenses/mit-license.php.
+
+#include "veriblock/algorithm.hpp"
+#include "veriblock/alt-util.hpp"
+#include "veriblock/entities/merkle_tree.hpp"
+
+namespace altintegration {
+
+uint256 CalculateContextInfoContainerHash(const PopData& popData,
+                                          const BlockIndex<AltBlock>& prevBlock,
+                                          const AltChainParams& params) {
+  auto firstPreviousKeystone =
+      prevBlock.getAncestor(getFirstPreviousKeystoneHeight(
+          prevBlock.getHeight() + 1, params.getKeystoneInterval()));
+  auto secondPreviousKeystone =
```

Comment:
this can be nullptr if ancestor is not found (for example VBK10->getAncestor(100) or VBK10->getAncestor(-100))

Reply:
If keystone is nullptr, then make it equal to `000..000`
```
auto zeroHash = prevBlock.getHash().fill(0);
```

Reply:
fixed